app/models/template_input.rb

```diff
@@ -0,0 +1,182 @@
+class TemplateInput < ApplicationRecord
+  include ::Exportable
+
+  class ValueNotReady < ::Foreman::Exception
+  end
+  class UnsatisfiedRequiredInput < ::Foreman::Exception
+  end
+
+  TYPES = { :user => N_('User input'), :fact => N_('Fact value'), :variable => N_('Variable'),
```

Comment:
This should be stored in a registry imho.
This makes it easier to read and easier to extend (from plugins).

Reply:
yeah, I tried to migrate 1:1 from REX to keep it minimal, if you wish, I can extend in this PR or in separate one (the same for separating input classes to different files)

Reply:
The registry can be added later, but I insist on the separate files. Separates files work a lot better with rails autoloading, I have seen strange issues when no separate files were used.

Reply:
ok, that's a good point, I haven't seen an issue with REX so far, but can imagine that autoloader would bite us in future, I'll split it as part of this PR, hopefully not a big deal for review